### DISCUSSION TITLE
How to handle state locking for multiple PRs opened at the same time

### DISCUSSION THREAD

**Discussion by florianmutter** [suzuki-shunsuke/tfaction#1533] [Deprecated (Question and Answer)]
### Question

We use renovate to update dependencies. Renovate creates multiple PRs at the same time, e.g. for updating different helm releases we deploy with terraform. tfaction starts to run at the same time resulting in failure because the state is locked.

I do see two options that I could think of that both have downsides
* Add  -lock-timeout=5m to terraform plan -> This could result in long running actions that don't do anything and jobs might still fail if there are to many open PRs
* Use the Github action concurrency feature to only allow one plan job per state -> Pending jobs are cancelled so in case of renovate opening 5 PRs we might end up with the first one and the last one running and all inbetween getting cancelled. See https://github.com/orgs/community/discussions/12835

Any suggestions what we could do?

### Example Code

GitHub Actions

```yaml

```

tfaction-root.yaml

```yaml

```

tfaction.yaml

```yaml

```

**Comment by suzuki-shunsuke** [comment]
tfaction itself doesn't provide any features and solutions about the State Locking issue,
but I think `-lock-timeout` and GitHub Actions concurrency are reasonable.

About Renovate, maybe you can use grouping feature to update multiple dependencies in the same pull request.
And maybe you can also change the schedule per dependency to prevent multiple pull requests from being created at the same time.

**Reply by suzuki-shunsuke** [reply]
This is a general issue of Terraform, so you would be able to ask help at the Terraform community.
